### DISCUSSION TITLE
Safari extension: Search feature and/or sorting

### DISCUSSION THREAD

**Discussion by JollyRoger8X** [simple-login/app#2143] [Feature Requests]
I have the SimpleLoigin version 1.3 Safari extension installed.

Often when I click the SimpleLogin toolbar widget, I want to find a simple login alias I've previously generated. Unfortunately, the sort order of the aliases in the list is not obvious (it is not alphabetical, nor can it be changed), and there is no search box where I can search to find the one I want. 

Please consider:

- adding a search function to the toolbar widget, allowing users to type in a few characters to filter the list, and 
- allowing users to specify the sort order of the alias list.

**Comment by nguyenkims** [comment]
Hi the latest safari version is rather 3.0.4, can you try with this version?

![CleanShot 2024-07-05 at 09 26 52@2x](https://github.com/simple-login/app/assets/3167780/241bea7d-e96a-4be7-9965-92724b8edcaa)

**Comment by JollyRoger8X** [comment]
This is confusing...

I originally installed the SimpleLogin Safari extension from the Apple App Store. 

But now I go to the App Store and search for SimpleLogin, and it seems to indicate this is a new app (it says **_Get_** as if I haven't installed it) and the version is **1.1.0**:

<img width="748" alt="image" src="https://github.com/simple-login/app/assets/6462046/b364b18d-58f6-45fc-8fd1-5d3a59484e74">
<img width="720" alt="image" src="https://github.com/simple-login/app/assets/6462046/addf92b5-4661-4c77-8fd1-74bc64852883">

I expected the App Store app would be updated automatically. It appears at some point Proton removed the old app and added a new one, unbeknownst to me!

So I deleted the old app, installed the new app, and even though the App Store says the current version is 1.1.0, it shows 3.0.4 in the Safari Preferences > Extensions window:

<img width="948" alt="image" src="https://github.com/simple-login/app/assets/6462046/827a346b-d00c-439b-90d4-8318903e8cf4">

When I enabled this extension, it opened a new browser tab to https://app.simplelogin.io/onboarding, which displays this message, but does nothing more:

<img width="548" alt="image" src="https://github.com/simple-login/app/assets/6462046/6c7b22e5-780d-4543-a14b-c66218c0e96e">

Running the app itself from /Applications just shows this window (note step 1 and step 2 say the exact same thing, which is confusing):

<img width="912" alt="image" src="https://github.com/simple-login/app/assets/6462046/78d8fa3e-a08c-4cbc-b8b6-ba4693ee3d91">

Apparently I am supposed to access the toolbar widget and sign in - I see there's a **Login with Proton** button at the bottom: 

<img width="458" alt="image" src="https://github.com/simple-login/app/assets/6462046/43d9abe5-e03e-4630-9381-4b494e4787b1">

But clicking **Login with Proton** and signing into my account just brings me to a new tab to https://app.simplelogin.io/onboarding/setup_done that says:

<img width="563" alt="image" src="https://github.com/simple-login/app/assets/6462046/d89b196e-36f0-4c4e-8c7a-6bbd9265b174">

And I am still not signed according to the toolbar widget:

<img width="464" alt="image" src="https://github.com/simple-login/app/assets/6462046/90151172-73fb-48cc-9f0a-c7250b9a4118">

What am I doing wrong here?

**Comment by JollyRoger8X** [comment]
I ended up giving up on logging in, and generating a new API key instead. That allowed me to log into the Safari extension.

Someone at Proton really needs to debug this process.

I am now running the latest version of the extension which does have a search feature. Thanks!

**Comment by nguyenkims** [comment]
There's indeed an issue with the login in the safari extension indeed, we're working on fix, sorry for the inconvenience! 

The reason that you don't have auto update is old extension and the new one are completely different apps. The new one is rewritten from ground up and use a different technology.
